### PR TITLE
chore: update .gitignore

### PR DESCRIPTION
**Pull request by datamweb**

### CODE REVIEW

**Review by paulbalandan — CHANGES_REQUESTED** [review]
Can you add a minimal gitignore file? Most of the changes here are IDE specific files. These should be added in a **global** gitignore file, not project-specific. If we are to add these, then all IDE files should be added here.

**Comment by datamweb** [comment]
@paulbalandan This file is a copy of the shield.
I don't know exactly which items you mean. Can you suggest a change?

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,107 @@
...
+
+# Linux
+*~
+
+# KDE directory preferences
+.directory
+
+# Linux trash folder which might appear on any partition or disk
+.Trash-*
+
```
Comment (paulbalandan) [inline .gitignore:38]: Should be in global gitignore.

File: .gitignore [hunk 2]
```diff
@@ -1,8 +1,107 @@
...
+#-------------------------
+# Environment Files
+#-------------------------
+# These should never be under version control,
+# as it poses a security risk.
+.env
+.vagrant
+Vagrantfile
+
```
Comment (paulbalandan) [inline .gitignore:47]: Also these.

File: .gitignore [hunk 3]
```diff
@@ -1,8 +1,107 @@
...
+#-------------------------
+# Temporary Files
+#-------------------------
+php_errors.log
```
Comment (paulbalandan) [inline .gitignore:51]: How to generate?

File: .gitignore [hunk 4]
```diff
@@ -1,8 +1,107 @@
...
+
+#-------------------------
+# Test Files
+#-------------------------
+tests/coverage*
+
```
Comment (paulbalandan) [inline .gitignore:57]: I don't think those are tests that should be ignored. Maybe this is in the wrong folder?

File: .gitignore [hunk 5]
```diff
@@ -1,8 +1,107 @@
...
+*.tmlanguage.cache
+*.tmPreferences.cache
+*.stTheme.cache
+*.sublime-workspace
+*.sublime-project
+.phpintel
+/api/
+
+# Visual Studio Code
+.vscode/
```
Comment (paulbalandan) [inline .gitignore:101]: IDE specific files should NOT be per project ignore file.

File: .gitignore [hunk 6]
```diff
@@ -1,8 +1,107 @@
...
+
+/results/
```
Comment (paulbalandan) [inline .gitignore:103]: what is this?

**Comment by kenjis** [comment]
> Most of the changes here are IDE specific files. These should be added in a global gitignore file, not project-specific. If we are to add these, then all IDE files should be added here.

What if contributors do not ignore these files?

**Comment by datamweb** [comment]
I was wondering how many people actually use global gitignore, I read about this today [here](https://gist.github.com/subfuzion/db7f57fff2fb6998a16c).
Do the codeigniter community really have the necessary knowledge of these issues? I think most people's answer is no.

**Comment by paulbalandan** [comment]
> What if contributors do not ignore these files?

Then I think it's time to educate them on these things.

**Comment by kenjis** [comment]
I don't use global gitignore. Because I have never needed it.
It doesn't seem so easy to actually try to set it up.

But I agree with making gitignore simpler for projects.
Why don't you prepare the documentation for it first?

**Comment by MGatner** [comment]
The Shield one is based off my Project template. I have a separate **.gitignore** for libraries that is specifically more minimalist: https://github.com/tattersoftware/codeigniter4-tools/blob/develop/src/Library/.gitignore

If these are desirable more generally we could incorporate them into DevKit. My Tools library basically spawned DevKit and then pared down to just a few things that didn't make it in.

**Comment by datamweb** [comment]
Well, I'll close this one, there are currently **.gitignore** file is minimalist.
If there is a need to change, anyone can do it.

**Pull request closed without merging**